cmd/google_cloud_ops_agent_diagnostics/main_others.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !windows
+// +build !windows
+
+package main
+
+import (
+	"flag"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/GoogleCloudPlatform/ops-agent/apps"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
+	"github.com/GoogleCloudPlatform/ops-agent/internal/self_metrics"
+	"github.com/shirou/gopsutil/host"
+)
+
+var (
+	config = flag.String("config", "/etc/google-cloud-ops-agent/config.yaml", "path to the user specified agent config")
+)
+
+func run() error {
+	_, mergedConfig, err := confgenerator.MergeConfFiles(*config, "linux", apps.BuiltInConfStructs)
+	if err != nil {
+		return err
+	}
+	hostInfo, err := host.Info()
+	if err != nil {
+		return err
+	}
+	uc, err := confgenerator.UnmarshalYamlToUnifiedConfig(mergedConfig, hostInfo.OS)
+	if err != nil {
+		return err
+	}
+
```

Comment:
Should the UnifiedConfig be validated? 
https://github.com/GoogleCloudPlatform/ops-agent/blob/52d2bd7a5ddebe3e6a721c79abb7bc7fe7ae7bb1/cmd/google_cloud_ops_agent_engine/main.go#L58-L61

Reply:
Yes, I wasn't sure if the validation was needed or not, since the `google_cloud_ops_agent_engine` binary would already generated the configs and verify them, but I decided to just put in a shared function in Linux and Windows with the all the validation steps.